pinot-core/src/main/java/org/apache/pinot/core/plan/AggregationPlanNode.java

```diff
@@ -118,6 +120,34 @@ public Operator<AggregationResultsBlock> buildNonFilteredAggOperator() {
     return new AggregationOperator(_queryContext, aggregationInfo, numTotalDocs);
   }
 
+  /**
+   * Returns {@code true} if any of the aggregation functions have null values, {@code false} otherwise.
+   *
+   * The current implementation is pessimistic and returns {@code true} if any of the arguments to the aggregation
+   * functions is of function type. This is because we do not have a way to determine if the function will return null
+   * values without actually evaluating it.
+   */
+  private boolean hasNullValues(AggregationFunction[] aggregationFunctions) {
+    for (AggregationFunction aggregationFunction : aggregationFunctions) {
+      ExpressionContext argument = (ExpressionContext) aggregationFunction.getInputExpressions().get(0);
+      switch (argument.getType()) {
+        case IDENTIFIER:
+          DataSource dataSource = _indexSegment.getDataSource(argument.getIdentifier());
+          NullValueVectorReader nullValueVector = dataSource.getNullValueVector();
+          if (nullValueVector != null && !nullValueVector.getNullBitmap().isEmpty()) {
+            return true;
+          }
+          break;
+        case LITERAL:
+          return false;
```

Comment:
This could potentially be the `null` literal value too but looks like this doesn't matter for both the cases? The non-scan based aggregation operator is only chosen if the aggregation function input operand is an identifier and for fast filtered count, it shouldn't matter because `COUNT(literal)` is treated the same as `COUNT(*)`.

Reply:
Hm, looks like Postgres and MySQL return `0` for `COUNT(null)` interestingly. We might want to do the same although I doubt it's a valid use case anyone cares about (and is also orthogonal to this PR).

Reply:
I guess we can transform this into: 
```java
          return argument.getLiteral().isNull();
```

Reply:
> The non-scan based aggregation operator is only chosen if the aggregation function input operand is an identifier and for fast filtered count, it shouldn't matter because COUNT(literal) is treated the same as COUNT(*)

But that is the fast literal and non scan limitation and responsibility. This functions returns whether there are nulls or not. If in the future we change the requirements say non scan aggregations, we shouldn't need to modify this code.

Reply:
Yeah, that makes sense 👍